src/pages/workspace/reimburse/WorkspaceReimburseView.js

```diff
@@ -84,6 +88,35 @@ class WorkspaceReimburseView extends React.Component {
         this.updateRateValueDebounced = _.debounce(this.updateRateValue.bind(this), 1000);
     }
 
+    componentDidMount() {
+        Policy.openWorkspaceReimburseView(this.props.policyID);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.policy.customUnits !== this.props.policy.customUnits) {
+            const distanceCustomUnit = _.chain(lodashGet(this.props, 'policy.customUnits', []))
+                .values()
+                .findWhere({name: CONST.CUSTOM_UNITS.NAME_DISTANCE})
+                .value();
+
+            this.setState({
+                unitID: lodashGet(distanceCustomUnit, 'customUnitID', ''),
+                unitName: lodashGet(distanceCustomUnit, 'name', ''),
+                unitValue: lodashGet(distanceCustomUnit, 'attributes.unit', 'mi'),
+                rateID: lodashGet(distanceCustomUnit, 'rates[0].customUnitRateID', ''),
+                rateName: lodashGet(distanceCustomUnit, 'rates[0].name', ''),
+                rateValue: this.getRateDisplayValue(lodashGet(distanceCustomUnit, 'rates[0].rate', 0) / 100),
+            });
+        }
```

Comment:
NAB: Why don't we use the props directly? I think it will be a bit hard to keep props and state in sync if we add more props in the future, no?

Reply:
Yeah, I'm trying to have as little impact on this aspect of the page as possible and simply support the new API command. It's been worked on by a number of contributors based on what I've seen. I agree it could use some improvement in how it's written but I feel like that would be the domain of a follow-up issue.